main.go

```diff
@@ -34,7 +35,8 @@ type metricProcessor struct {
 	cfClientConfig *cfclient.Config
 	msgChan        chan *metrics.Stream
 	errorChan      chan error
-	watchedApps    map[string]*consumer.Consumer
+	watchedApps    map[string]chan cfclient.App
+	sync.RWMutex
 }
 
 func (tr metricProcessor) RefreshAuthToken() (token string, authError error) {
```

Comment:
gometalinter is complaining that this function passes a lock by value. Should we instead use `tr *metricProcessor` here?

Reply:
The rest of the file refers to the `metricProcessor` in the receiver position as `m` and not `tr`. Should we adjust?

Reply:
Apologies. This was an existing function - ignore for this PR.

Reply:
Yeah yuk ... but ignoring for now 😞